rest-api/src/main/java/io/quarkus/qe/SearchResource.java

```diff
@@ -26,19 +29,19 @@
 
     @Query("repositoryById")
     @Description("Get repository by ID")
-    public Repository getRepositoryByRepoUrl(long id) throws RepositoryNotFoundException {
+    public Repository getRepositoryByRepoUrl(@Name("id") long id) throws RepositoryNotFoundException {
         return repositoryService.findById(id);
     }
 
     @Query("repositoryByUrl")
     @Description("Get repository by repository URL")
-    public Repository getRepositoryByRepoUrl(String repoUrl) throws RepositoryNotFoundException {
+    public Repository getRepositoryByRepoUrl(@Name("repoUrl") String repoUrl) throws RepositoryNotFoundException {
         return repositoryService.findByRepoUrl(repoUrl);
     }
 
     @Query("repositoriesByExtensions")
     @Description("Get repositories using a list of extensions")
-    public List<Repository> getRepositoriesByExtension(List<String> extensions) {
+    public List<Repository> getRepositoriesByExtension(@Name("extensions") List<ExtensionsQuery> extensions) {
```

Comment:
Because I think that is not a very good idea to push our model to an endpoint or a database. I mean, your model must be decoupled to any port or layer. Otherwise if the model change, we will break the contract (API).

Reply:
This model is intended to be shared. Indeed, our REST API uses it.